cmake/FindASIO.cmake

```diff
@@ -0,0 +1,33 @@
+find_path(ASIO_INCLUDE_DIR
+    NAMES asio/version.hpp
+)
+
+mark_as_advanced(ASIO_INCLUDE_DIR)
+
+if(ASIO_INCLUDE_DIR AND EXISTS "${ASIO_INCLUDE_DIR}/asio/version.hpp")
+    file(STRINGS "${ASIO_INCLUDE_DIR}/asio/version.hpp" _asio_version_line
+         REGEX "^#[ \t]*define[ \t]+ASIO_VERSION[ \t]+([0-9]+).*")
+    string(REGEX REPLACE "^.*ASIO_VERSION[ \t]+([0-9]+).*$" "\\1"
+           _asio_version "${_asio_version_line}")
+    math(EXPR _asio_major "${_asio_version} / 100000")
+    math(EXPR _asio_minor "(${_asio_version} / 100) % 1000")
+    math(EXPR _asio_patch "${_asio_version} % 100")
+    set(ASIO_VERSION "${_asio_major}.${_asio_minor}.${_asio_patch}")
+endif()

```

Comment:
Wow that's complicated... And unused except for probably a log message.

Reply:
Thanks, that reminds me I forgot to add a minimum version check ;)

Reply:
This explains much 😃